### PR TITLE
[wallet-ext] bullsharks interstitial

### PR DESCRIPTION
**Pull request by Nikhil-Mysten**
## Description 

image
image

## Test Plan 

Locally

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/index.tsx [hunk 1]
```diff
@@ -57,10 +67,28 @@ const App = () => {
 	}, [location, dispatch]);
 
 	useInitialPageView();
+	useEffect(() => {
+		(async () => {
+			const interstitialDismissed = await get<boolean>('bullshark-interstitial-dismissed');
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/index.tsx:72]: Can you just use localstorage for this value instead?

File: apps/wallet/src/ui/styles/global.scss [hunk 2]
```diff
@@ -105,3 +105,11 @@ textarea {
 ::-ms-reveal {
 	display: none;
 }
+
+@font-face {
+	font-family: 'Frankfurter Normal';
+	src: url('../assets/images/FrankfurterNormal.woff2') format('woff2'),
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/styles/global.scss:111]: In images folder but not an image?

File: apps/wallet/src/ui/app/pages/home/bullshark-quests-notification/Overlay.module.scss [hunk 3]
```diff
@@ -0,0 +1,31 @@
+@use '_variables' as v;
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/bullshark-quests-notification/Overlay.module.scss:1]: Don't introduce new cSS modules even for quick projects.

File: apps/wallet/src/ui/app/index.tsx [hunk 4]
```diff
@@ -57,10 +67,28 @@ const App = () => {
...
+			if (interstitialDismissed) {
+				setInternalInterstitialDismissed(interstitialDismissed);
+			}
+		})();
+	}, []);
 
 	return (
 		<Routes>
 			<Route path="/*" element={<HomePage />}>
+				<Route
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/index.tsx:82]: why is this a route? Seems like it really should just be an overlay on the tokens page?

**Review by Jordan-Mysten — CHANGES_REQUESTED** [review]

File: apps/wallet/src/ui/app/pages/home/bullshark-quests-notification/index.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+							<div className="bg-[url('https://quests.mystenlabs.com/_next/static/media/logo.81b4eb8f.svg')] h-40 w-40 bg-cover" />
+							<InterstitialHeading text="5 Million SUI prize pool!" />
+							<div className="flex flex-col items-center gap-2 mt-2">
+								<InterstitialBanner lines={['2.5M SUI', 'Top 10,000 players!']} />
+								<InterstitialBanner lines={['2.5M SUI', 'all eligible players!']} />
+							</div>
+						</div>
+
+						<div className="flex flex-col items-center gap-4 [-webkit-text-stroke:1px_black] w-full mt-5">
+							<a
```
Comment (plam-ml) [inline apps/wallet/src/ui/app/pages/home/bullshark-quests-notification/index.tsx:43]: Do we need `target="_blank"` for these a tags? I see an `ExternalLink` component, and wonder if we should be using that instead?

**Review by plam-ml — APPROVED** [review]
Some Qs, but overall lgtm

File: apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx [hunk 6]
```diff
@@ -200,6 +207,21 @@ function TokenDetails({ coinType }: TokenDetailsProps) {
 	// Avoid perpetual loading state when fetching and retry keeps failing add isFetched check
 	const isFirstTimeLoading = isLoading && !isFetched;
 
+	useEffect(() => {
+		const dismissed = localStorage.getItem('bullshark-interstitial-dismissed');
+		setInterstitialDismissed(Boolean(dismissed));
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx:212]: `Boolean('false') === true` so probably don't want to do it this way. Probably just check the value directly (dismissed === 'true')

**Review by Jordan-Mysten — APPROVED** [review]

**Pull request merged**